<launch>   

  <!--
  Terminal 1. roscore
  Terminal 2. rosparam set use_sim_time true && roslaunch pff velodyne_bag_slam.launch
  Terminal 3. rosbag play - -clock [BAG_NAME]
  -->

  <!-- Start filter -->
  <node name="velodyne_filter" pkg="traversable_mapping" type="velodyne_filter">
    <param name="topicSubscriber" value="/velodyne_points"/> 
    <param name="topicPublisher" value="/scan"/> 
    <param name="frameID" value="base_link"/> 
    <param name="robotHeight" value="0.85"/> 
    <param name="sensorHeight" value="0.74"/> 
    <param name="resolution" value="0.2"/> 
    <param name="visionAngle" value="360"/> 
  </node>

  <!-- start gmapping -->
  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping">
    <remap from="scan" to="scan"/>
    <param name="map_update_interval" value="5.0"/>      
    <param name="linearUpdate" value="0.0"/>
    <param name="angularUpdate" value="0.0"/>
  </node>	

  <!-- start rviz -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find traversable_mapping)/rviz_cfg/slam.rviz" />

</launch>
